File: server/Services/Restore/Restore.Core/Entities/OrderAggregate/Order.cs
namespace Restore.Core.Entities.OrderAggregate;

public class Order
{
    public int Id { get; set; }
    public string BuyerId { get; set; }

    public ShippingAddress ShipToAddress { get; set; }
    public DateTime OrderDate { get; set; }
    public List<OrderItem> OrderItems { get; set; }
    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;

    public long GetTotal()
    {
        return Subtotal + DeliveryFee;
    }
}

File: client/src/app/models/order.ts
export interface IShippingAddress {
  fullName: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  zip: string;
  country: string;
}

export interface IOrderItem {
  productId: number;
  name: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}

export interface IOrder {
  id: number;
  buyerId: string;
  orderDate: string;
  shippingAddress: IShippingAddress;
  deliveryFee: number;
  orderItems: IOrderItem[];
  subtotal: number;
  orderStatus: string;
  total: number;
}
